en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/tls/component-certificates.md

```diff
@@ -0,0 +1,91 @@
+- Although the router also has its truststore and keystore in the above mentioned locations, there is a special case when the above locations will not be referred. This is when connecting to backend services exposed by the APIs. For more information on how to add certificates of backend services, click the link [here]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/security/backend-certificates).
+
+Since the certificates are used for different purposes, the certificate location referred for each purpose can be changes by editing the `config.toml` and the relevant docker container volume mounts.
+
+## Adding a certificate to a component truststore
+
+To add a new certificate to a Choreo Connect component, the pem formatted certificate must be added to the *truststore* location of that particular component.
+
+1.  Convert the public certificate to a PEM format. For example,
+
+    `openssl x509 -inform der -in public_certificate.cert -out certificate.pem`
+
+2.  Add the certificate to the relevant component's resource folder
+
+    ``` tab="adapter"
+    <CHOREO-CONNECT_HOME>/docker-compose/resources/adapter/security/truststore
+    ```
+
+    ``` tab="enforcer"
+    <CHOREO-CONNECT_HOME>/docker-compose/resources/enforcer/security/truststore
+    ```
+
+    ``` tab="router"
+    <CHOREO-CONNECT_HOME>/docker-compose/resources/router/security/truststore
+    ```
+        
+3.  Restart the component.
+   
+    `docker restart <container_name>`
+   
+
+!!! info
+
+    The certificate locations are configured as a volume mount for each component container in the `docker-compose.yaml` file as given below.
+
+    ``` tab="adapter"
+    volumes:
+            - ../resources/adapter/security:/home/wso2/security
+    ```
+
+    ``` tab="enforcer"
+    volumes:
+            - ../resources/enforcer/security:/home/wso2/security
+    ```
+
+    ``` tab="router"
+    volumes:
+            - ../resources/router/security:/home/wso2/security
+    ```
+
+    This mounts both the key store and truststore locations to the container.
+
+
+## Changing the Private Certificate of a Component
+If it is required to change the private certificate of a component, follow the steps below.
+
+1. Generate a new key pair for the component.
+   
+2. Copy the private key and certificate (in PEM format) into `<CHOREO-CONNECT_HOME>/resources/<COMPONENT>/security/keystore` directory.
+
+    Ex: To change the private key of Adapter component, copy the new certificate to `<CHOREO-CONNECT_HOME>/resources/adapter/security/keystore` directory.
+
+3. If the private key file name is changed, edit the corresponding configuration in `<CHOREO-CONNECT_HOME>/docker-compose/choreo-connect/conf/config.toml` or `<CHOREO-CONNECT_HOME>/docker-compose/choreo-connect-with-apim/conf/config.toml` file. Otherwise, use the same name as `mg.pem` and `mg.key` for the new certificate and key.
```

Comment:
```suggestion
3. If you change the private key file name, edit the corresponding configuration in the `<CHOREO-CONNECT_HOME>/docker-compose/choreo-connect/conf/config.toml` or `<CHOREO-CONNECT_HOME>/docker-compose/choreo-connect-with-apim/conf/config.toml` file. Otherwise, use the same name as `mg.pem` and `mg.key` for the new certificate and key.
```